test/qa.jl

```diff
@@ -0,0 +1,13 @@
+using DelayDiffEq, Aqua
+@testset "Aqua" begin
+    Aqua.find_persistent_tasks_deps(DelayDiffEq)
+    Aqua.test_ambiguities(DelayDiffEq, recursive = false)
+    Aqua.test_deps_compat(DelayDiffEq)
+    Aqua.test_piracies(DelayDiffEq,
+        treat_as_own = [DelayDiffEq.SciMLBase.DESolution,
+        DelayDiffEq.SciMLBase.SciMLSolution])
+    Aqua.test_project_extras(DelayDiffEq)
+    Aqua.test_stale_deps(DelayDiffEq)
+    Aqua.test_unbound_args(DelayDiffEq)
+    Aqua.test_undefined_exports(DelayDiffEq)
```

Comment:
Again, I'd prefer to use `Aqua.test_all` to not miss out on any tests in the future (again, it seems there's some fine-tuning with keyword arguments necessary anyway).
Since the ambiguity check is problematic currently, I suggest (as e.g. in Distributions and Pumas) using
```suggestion
    # Test ambiguities separately without Base and Core
    # Ref: https://github.com/JuliaTesting/Aqua.jl/issues/77
    Aqua.test_all(DelayDiffEq; ambiguities = false)
    Aqua.test_ambiguities(DelayDiffEq)
```